File: 71-simplify-path/71-simplify-path.java
class Solution {
    public String simplifyPath(String path) {
        String[] splittedPath = path.split("/");
        Stack<String> stack = new Stack<>();
        
        for (String s: splittedPath) {
            
            if (s.equals("") || s.equals(".") ) {
                continue;
            } else if (s.equals("..")) {
                if (stack.size() > 0) {
                    stack.pop();
                };
            } else {
                stack.push(s);
            };
            
        };
        
        if (stack.isEmpty()) return "/";
        
        StringBuilder ans = new StringBuilder("");
        for (String s: stack) {
            ans.append('/');
            ans.append(s);
            
        };
        
        return ans.toString();
        
    }
}

File: 71-simplify-path/71-simplify-path.py
class Solution:
    def simplifyPath(self, path: str) -> str:
        path = path.split("/")
        stack = []
        
        for i in path:
            if i == '' or i == '.':
                continue
            elif i == '..':
                if len(stack):
                    stack.pop()
            else:
                stack.append(i)
         
        return "/" + "/".join(stack)